### ISSUE TITLE
[REQUEST] Custom terminal window title like BashTOP and BPyTOP

### ISSUE THREAD

**Issue by RedBearAK** [aristocratos/btop#198]
Both BashTOP and BPyTOP were able to change the title of the terminal window to reflect that they were actively running in that window. This is obviously a very nice feature to have when switching between multiple terminal windows. BTOP doesn't seem to have this feature re-implemented yet. I hope it won't be difficult to implement. I think I was even the one who asked for it to be re-implemented in BPyTOP while you were in the process of porting all the features from BashTOP, and you were able to do it almost immediately in that case. 

I was a bit shocked just now to install HTOP (for comparison, since I think it's written in C++) and find that it doesn't change the window title if you just run it from the CLI. Even when I start HTOP from the GNOME (Ubuntu 21.10) app launcher as if it is a GUI application, the window is just titled "Terminal". Lame. 

So if you could implement this in BTOP the way it works in BashTOP and BPyTOP it would be a real stand-out feature.

**Comment by aristocratos** [comment]
@RedBearAK 
Yeah, shouldn't be an issue to add. Mostly just forgot to add it in the first place...

**Comment by RedBearAK** [comment]
@aristocratos 

>  Yeah, shouldn't be an issue to add. Mostly just forgot to add it in the first place...

I figured. Looking forward to the next release. It's coming along quite nicely already.

**Comment by Ptitet** [comment]
Hey, do you have any update on this feature ? As of today (1.4.4), btop does not change the window title. Thanks !

**Comment by deckstose** [comment]
What would you expect from such this feature? I've just implemented title changing to `btop` and I don't think this adds any value over the previous title, the command + cwd.

**Comment by Ptitet** [comment]
> What would you expect from such this feature?
Being able to automatically maximize my terminal when I launch `btop`, and its nice to have windows' titles correctly set, especially when using a quick switcher between open windows (alt-tab).

I am on NixOS with btop from Nixpkgs unstable (so lastest version of btop) and it doesn't change the title of the window when launched. I tested with zsh, bash and sh.

**Comment by sim590** [comment]
I'm also interested in this feature. Like @Ptitet, I use a tiling window manager (Hyprland) and the terminal title is important for window identification, both in the window switcher and for window rules (auto-maximize, placement, etc.).

Currently, btop doesn't send any OSC 2 escape sequence, so the terminal title stays at whatever the shell or terminal emulator sets by default (e.g. "Alacritty" or just the cwd). This is especially noticeable when btop is launched directly by the WM rather than from an interactive shell, since there's no shell prompt to set the title.

Even a simple hardcoded `btop` title would be a meaningful improvement over the current behavior. A configurable title string would be even better (addressing #1246 as well).

@deckstose Since you mentioned you already have an implementation, would you consider submitting it as a PR? It would be useful to a number of us.
